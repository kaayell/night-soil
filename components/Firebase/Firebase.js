import * as firebase from 'firebase'
import { ANDROID_API_KEY, config, IOS_API_KEY } from './secrets'

export default class Firebase {
  static init () {
    firebase.initializeApp(config)
  }

  static getAuth() {
    return firebase.auth()
  }

  static getUserDetailsRef() {
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/user/details/${uid}`)
  }

  static savePoop(poopDetails) {
    let currentUser = firebase.auth().currentUser
    firebase.database().ref(`/poop/${currentUser.uid}`).push().set(poopDetails)
  }

  static getPoopsRef(){
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/poop/${uid}`)
  }

  static async loginWithGoogle () {
    const response = await Expo.Google.logInAsync({
      androidClientId: ANDROID_API_KEY,
      iosClientId: IOS_API_KEY,
      scopes: ['profile', 'email'],
    })

    if (response.type === 'success') {
      const credential = firebase.auth.GoogleAuthProvider.credential(response.idToken, response.accessToken)
      firebase.auth().signInWithCredential(credential)
    }
  }
}