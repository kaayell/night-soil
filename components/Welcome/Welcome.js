import React from 'react'
import { Image, Text, View } from 'react-native'
import Firebase from '../Firebase/Firebase'
import { BLUE, OFF_WHITE } from '../StyleGuide/colors'
import { POPPINS } from '../StyleGuide/fonts'
import { Button } from 'react-native-elements'

export const Welcome = () => {
  return (
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center', backgroundColor: BLUE}}>
      <Image source={require('../../assets/triangle.png')}
             style={{width: 150, height: 150}}/>
      <Text style={{color: OFF_WHITE, fontFamily: POPPINS, fontSize: 30}}>NIGHT SOIL</Text>
      <Button
        title='SIGN IN WITH GOOGLE'
        onPress={() => Firebase.loginWithGoogle()}
        textStyle={{color: OFF_WHITE, fontWeight: '700'}}
        buttonStyle={{
          marginTop: 10,
          backgroundColor: 'transparent',
          width: 300,
          height: 45,
          borderColor: OFF_WHITE,
          borderWidth: .5,
          borderRadius: 5
        }}
      />
    </View>)
}