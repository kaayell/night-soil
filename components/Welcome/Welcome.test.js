import {shallow} from 'enzyme'
import React from 'react'
import {Welcome} from './Welcome'
import Firebase from '../Firebase/Firebase'
import {Button} from "react-native-elements";

describe('Welcome', () => {
  it('should render', () => {
    let component = shallow(<Welcome />)
    expect(component).toMatchSnapshot()
  })

  it('should call firebase on press of button', () => {
    Firebase.loginWithGoogle = jest.fn()
    let component = shallow(<Welcome />)
    component.find(Button).simulate('press')
    expect(Firebase.loginWithGoogle).toHaveBeenCalled()
  })
})