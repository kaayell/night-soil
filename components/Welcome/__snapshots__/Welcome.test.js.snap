// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9AC0CD",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/poop-white.png",
      }
    }
    style={
      Object {
        "height": 150,
        "width": 150,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FAFAFA",
        "fontFamily": "poppins-regular",
        "fontSize": 30,
      }
    }
  >
    NIGHT SOIL
  </Text>
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#FAFAFA",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "height": 45,
        "width": 300,
      }
    }
    onPress={[Function]}
    textStyle={
      Object {
        "color": "#FAFAFA",
        "fontWeight": "700",
      }
    }
    title="SIGN IN WITH GOOGLE"
  />
</View>
`;
